.github/workflows/release.yaml

```diff
@@ -0,0 +1,280 @@
+        name: Create the full version release
+        run: |
+          gh release create ${{ needs.extract-version-details.outputs.full-version }} \
+            --draft \
+            --prerelease=${{ inputs.kind != 'production' }} \
+            --generate-notes \
+            --title "Captain ${{ needs.extract-version-details.outputs.full-version }}"
+      - name: Update the release if it already exists
+        if: ${{ steps.create-full-version-release.outcome == 'failure' }}
+        run: gh release edit ${{ needs.extract-version-details.outputs.full-version }} --tag ${{ needs.extract-version-details.outputs.full-version }}
+
+  build-and-upload-binaries:
+    runs-on: ubuntu-latest
+    needs:
+      - extract-version-details
+      - draft-full-version-release
+    permissions:
+      contents: write
+      id-token: write
+    strategy:
+      matrix:
+        os: [linux, darwin, windows]
+        arch: [amd64, arm64]
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.extract-version-details.outputs.full-version }}
+      - uses: cachix/install-nix-action@v18
+        with:
+          extra_nix_config: |
+            keep-derivations = true
+            keep-outputs = true
+      - run: |
+          GOOS=${{ matrix.os }} \
+            GOARCH=${{ matrix.arch }} \
+            CGO_ENABLED=0 \
+            LDFLAGS="-w -s -X github.com/rwx-research/captain-cli.Version=${{ needs.extract-version-details.outputs.full-version }}" \
+            nix develop --command mage
+      - if: ${{ matrix.os != 'windows' }}
+        run: |
+          if [[ "${{ matrix.arch }}" == "amd64" ]]; then
+            new_arch="x86_64"
+          elif [[ "${{ matrix.arch }}" == "arm64" ]]; then
+            new_arch="aarch64"
+          else
+            echo "unmapped arch: ${{ matrix.arch }}"
+            exit 1
+          fi
+
+          github_asset_name=$(echo "captain-${{ matrix.os }}-$new_arch" | tr '[:upper:]' '[:lower:]')
+          mv captain "$github_asset_name"
+          gh release upload ${{ needs.extract-version-details.outputs.full-version }} "$github_asset_name" --clobber
+      - if: ${{ matrix.os == 'windows' }}
+        run: |
+          if [[ "${{ matrix.arch }}" == "amd64" ]]; then
+            new_arch="x86_64"
+          elif [[ "${{ matrix.arch }}" == "arm64" ]]; then
+            new_arch="aarch64"
+          else
+            echo "unmapped arch: ${{ matrix.arch }}"
+            exit 1
+          fi
+
+          github_asset_name=$(echo "captain-${{ matrix.os }}-$new_arch.exe" | tr '[:upper:]' '[:lower:]')
+          mv captain.exe "$github_asset_name"
+          gh release upload ${{ needs.extract-version-details.outputs.full-version }} "$github_asset_name" --clobber
+
+  obtain-approval:
+    runs-on: ubuntu-latest
+    if: ${{ inputs.kind == 'production' }}
+    needs:
+      - build-and-upload-binaries
+    environment: release-approval
```

Comment:
that's a neat trick. I didn't realize you could do that.

Reply:
(presumably at this stage you edit the release notes? before approval)

Reply:
Yep! (or after, I guess, but probably at this point)